### PR TITLE
Andrey Rohau / GetTest For Coach

### PR DESCRIPTION
**Pull request by andreyrogov8**
removed: Extra libraries, Extra fields,
Added: Validations, Search filters, Default value for Priority

### CODE REVIEW

File: Elevel/Application/Features/QuestionCommands/GetQuestionDetailQuery.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public Handler(IApplicationDbContext context, IMapper mapper)
 
             public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
             {
-                var question = await _context.Questions.AsNoTracking()
+                var question = await _context.Questions.AsNoTracking().Where(x => !x.AuditionId.HasValue)
```
Comment (parys) [inline Elevel/Application/Features/QuestionCommands/GetQuestionDetailQuery.cs:35]: it's not a bug, but it's better to keep 1 linq operator per row

File: Elevel/Application/Features/TestCommands/CheckTestCommand.cs [hunk 2]
```diff
@@ -31,6 +30,20 @@ public class Request : IRequest<Response>
             public Guid CoachId { get; set; }
         }
 
+        public class Validator : AbstractValidator<Request>
+        {
+
+            private const int ANSWER_COUNT = 4;
```
Comment (parys) [inline Elevel/Application/Features/TestCommands/CheckTestCommand.cs:36]: for which reason you have const here?

File: Elevel/Application/Features/TestCommands/AssignTestCommand.cs [hunk 3]
```diff
@@ -23,12 +21,25 @@ public class Request : IRequest<Response>
...
+            public bool Priority { get; set; } = false;
 
             [JsonIgnore]
             public Guid HrId { get; set; }
         }
 
+        public class Validator : AbstractValidator<Request>
+        {
+
+            private const int ANSWER_COUNT = 4;
```
Comment (parys) [inline Elevel/Application/Features/TestCommands/AssignTestCommand.cs:33]: for which reason you have const here?

File: Elevel/Application/Features/TestCommands/StartTestByIdQuery.cs [hunk 4]
```diff
@@ -28,6 +28,15 @@ public class Request : IRequest<Response>
             [JsonIgnore]
             public Guid UserId { get; set; }
         }
+        public class Validator : AbstractValidator<Request>
+        {
+
+            private const int ANSWER_COUNT = 4;
```
Comment (parys) [inline Elevel/Application/Features/TestCommands/StartTestByIdQuery.cs:34]: for which reason you have const here?

File: Elevel/Application/Features/QuestionCommands/GetQuestionDetailQuery.cs (hunk 1 above)
Comment (DShcharbakou) [reply Elevel/Application/Features/QuestionCommands/GetQuestionDetailQuery.cs]: this is necessary when the line becomes unreadable IMHO

**Review by DShcharbakou — APPROVED** [review]

**Review by parys — APPROVED** [review]

**Pull request merged**